src/components/FlowRunArtifacts.vue

```diff
@@ -0,0 +1,131 @@
+    <template v-if="hasArtifacts">
+      <div class="flow-run-artifacts__button-group-container">
+        <slot name="actions" />
+        <ViewModeButtonGroup class="flow-run-artifacts__view-mode-button-group" />
+      </div>
+
+      <p-heading heading="6" class="flow-run-artifacts__subheading">
+        {{ localization.info.flowRun }}
+      </p-heading>
+
+      <RowGridLayoutList :items="flowRunArtifacts">
+        <template #default="{ item }">
+          <router-link :to="routes.artifact(item.id)">
+            <ArtifactCard :artifact="item" :condense="condense" class="flow-run-artifacts__artifact" />
+          </router-link>
+        </template>
+
+        <template #empty>
+          <div class="flow-run-artifacts__none">
+            <p-markdown-renderer :text="localization.info.none" />
+          </div>
+        </template>
+      </RowGridLayoutList>
+
+      <p-divider />
+
+      <p-heading heading="6" class="flow-run-artifacts__subheading">
+        {{ localization.info.taskRuns }}
+      </p-heading>
+
+      <RowGridLayoutList :items="taskRunArtifacts">
+        <template #default="{ item }">
+          <router-link :to="routes.artifact(item.id)">
+            <ArtifactCard :artifact="item" :condense="condense" class="flow-run-artifacts__artifact" />
+          </router-link>
+        </template>
+
+        <template #empty>
+          <div class="flow-run-artifacts__none">
+            <p-markdown-renderer :text="localization.info.none" />
+          </div>
+        </template>
+      </RowGridLayoutList>
+    </template>
+
+    <template v-else-if="artifactsSubscription.executed">
+      <p-empty-state>
+        <template #description>
+          <p-markdown-renderer :text="emptyMessage" />
+        </template>
+      </p-empty-state>
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { useSubscription } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import ArtifactCard from '@/components/ArtifactCard.vue'
+  import RowGridLayoutList from '@/components/RowGridLayoutList.vue'
+  import ViewModeButtonGroup from '@/components/ViewModeButtonGroup.vue'
+  import { useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
+  import { localization } from '@/localization'
+  import { FlowRun, isTerminalStateType } from '@/models'
+  import { ArtifactsFilter } from '@/models/Filters'
+  import { activeViewMode } from '@/utilities/activeViewMode'
+
+  const props = defineProps<{
+    flowRun: FlowRun,
+  }>()
+
+  const condense = computed(() => activeViewMode.value !== 'grid')
+
+  const api = useWorkspaceApi()
+  const routes = useWorkspaceRoutes()
+
+  const artifactsFilter = computed<ArtifactsFilter>(() => {
+    return {
+      artifacts: {
+        flowRunId: [props.flowRun.id],
+        notType: ['result'],
+      },
+    }
+  })
+  const artifactsSubscription = useSubscription(api.artifacts.getArtifacts, [artifactsFilter], { interval: 5000 })
```

Comment:
Is it worth using `isTerminalStateType` to conditionally poll here?

Reply:
Or only poll when `isRunningStateType` which is what we did for the timeline.

Reply:
Yeah it definitely is - [ticket here](https://github.com/PrefectHQ/prefect-ui-library/issues/1320) so I can handle that in a follow up (will want to do that for results as well)